app/hooks/useImportFileWarning.ts

```diff
@@ -0,0 +1,65 @@
+import { useState, useEffect } from 'react';
+import { ipcRenderer } from 'electron';
+import { ipcConsts } from '../vars';
+import { ImportWalletWarningRequest } from '../../shared/ipcMessages';
+import { eventsService } from '../infra/eventsService';
+
+type UseImportFileWarningReturn = {
+  isOpen: boolean;
+  message: string;
+  handleResponse: (userResponse: boolean) => void;
+};
+
+const useImportFileWarning = (): UseImportFileWarningReturn => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [message, setMessage] = useState<string>('');
+
+  useEffect(() => {
+    const handleShowModal = (
+      _: Electron.IpcRendererEvent,
+      payload: ImportWalletWarningRequest
+    ) => {
+      const { isDuplicateName, isDuplicateWallet, isDuplicatePath } = payload;
+
+      const similarities: string[] = [];
+      if (isDuplicatePath) {
+        similarities.push('file path');
+      }
+      if (isDuplicateName) {
+        similarities.push('name');
+      }
+      if (isDuplicateWallet) {
+        similarities.push('encrypted data');
+      }
+
+      let formattedMessage = 'A wallet with the same ';
+      if (similarities.length > 0) {
+        formattedMessage += similarities.join(', ');
+      } else {
+        formattedMessage = 'A wallet';
+      }
+      formattedMessage += ' already exists.';
```

Comment:
@brusherru will for sure assess it better. I don’t have proper knowledge, but to me, it seems this could be improved. The final message does not sound really well and it's hard to correct. 
My suggestion (keep in mind that I know almost nothing about the TS/JS.
```suggestion
      const { isDuplicateName, isDuplicateWallet, isDuplicatePath } = payload;

    let formattedMessage = '';

    if (isDuplicateWallet || isDuplicatePath) {
      // If the wallet or/and path is duplicated, regardless of name duplication
      formattedMessage = "This wallet is already opened. Do you want to import it anyway?";
    } else if (isDuplicateName) {
      // If only the name is duplicated, the path or encrypted data different
      formattedMessage = "A different wallet with the same name is already opened in Smapp. Double-check which one you use and consider renaming one of them.";
    }
```